libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1044,15 +1045,53 @@ TrackListHolder WaveTrackFactory::Create(size_t nChannels)
    return Create(nChannels, QualitySettings::SampleFormatChoice(), mRate.GetRate());
 }
 
+void WaveTrack::MoveAndSwapAttachments(WaveTrack &&other)
+{
+   this->AttachedTrackObjects::ForCorresponding(other,
+   [](TrackAttachment *pLeft, TrackAttachment *pRight){
+      // Precondition of callback from ClientData::Site
+      assert(pLeft && pRight);
+      const auto pLeftAttachments =
+         dynamic_cast<ChannelAttachmentsBase *>(pLeft);
+      const auto pRightAttachments =
+         dynamic_cast<ChannelAttachmentsBase *>(pRight);
+      // They should have come from the same factory of channel attachments
+      assert((pLeftAttachments == nullptr) == (pRightAttachments == nullptr));
+      if (pLeftAttachments) {
+         *pLeftAttachments = std::move(*pRightAttachments);
+         pLeftAttachments->SwapChannels();
+      }
+   });
+}
+
+void WaveTrack::MergeChannelAttachments(WaveTrack &&other)
+{
+   this->AttachedTrackObjects::ForCorresponding(other,
+   [](TrackAttachment *pLeft, TrackAttachment *pRight){
+      // Precondition of callback from ClientData::Site
+      assert(pLeft && pRight);
+      const auto pLeftAttachments =
+         dynamic_cast<ChannelAttachmentsBase *>(pLeft);
+      const auto pRightAttachments =
+         dynamic_cast<ChannelAttachmentsBase *>(pRight);
+      // They should have come from the same factory of channel attachments
+      assert((pLeftAttachments == nullptr) == (pRightAttachments == nullptr));
+      if (pLeftAttachments)
+         pLeftAttachments->MakeStereo(std::move(*pRightAttachments));
+   });
+}
+
 TrackListHolder WaveTrackFactory::Create(size_t nChannels, sampleFormat format, double rate)
 {
-   auto channels = std::vector<std::shared_ptr<Track>>{ };
-   std::generate_n(
-      std::back_inserter(channels),
+   auto channels = std::vector<std::shared_ptr<WaveTrack>>{ };
+   generate_n(
```

Comment:
ADL is a lot of fun, but I wonder if it generates new warnings from Clang. Right now, ADL for `std::move` used across several of your PRs generates enormous amounts of warnings, making it very hard to diagnose macOS.

If it generates new warnings, please do not use ADL here.

Reply:
For `std::move`, LLVM guys consider it harmful because `move` is a common name. I don't know if it applies to any of the functions in this case.

Reply:
What are examples of the warnings?

Reply:
There are 2486 warnings with message
```
unqualified call to 'std::move' [-Wunqualified-std-cast-call]
```
This is roughly 60% of the warnings Clang generates.

Reply:
They multiply quickly because `Registry.h` contains two places, that generate this warning.